### PR TITLE
test(fw): add empty account tests

### PR DESCRIPTION
**Pull request by marioevz**
## 🗒️ Description
Fixes empty accounts not being recognized as such because an explicit zero was set as the value to a key.

Edit(dan): The original intent of this PR was also fixed in #486 and this PR now only adds additional tests for empty accounts and one small refactor.

## 🔗 Related Issues
Fixes https://github.com/ethereum/execution-spec-tests/issues/475

## ✅ Checklist
- [x] All: Set appropriate labels for the changes.
- [x] All: Considered squashing commits to improve commit history.
- [x] All: Added an entry to [CHANGELOG.md](/ethereum/execution-spec-tests/blob/main/docs/CHANGELOG.md).
- [x] All: Considered updating the online docs in the [./docs/](/ethereum/execution-spec-tests/blob/main/docs/) directory.
- [x] Tests: All converted JSON/YML tests from [ethereum/tests](/ethereum/tests) have been added to [converted-ethereum-tests.txt](/ethereum/execution-spec-tests/blob/main/converted-ethereum-tests.txt).
- [x] Tests: Included the type and version of evm t8n tool used to locally execute test cases:  e.g., ref with commit hash or geth 1.13.1-stable-3f40e65.
- [x] Tests: Ran `mkdocs serve` locally and verified the auto-generated docs for new tests in the [Test Case Reference](https://ethereum.github.io/execution-spec-tests/main/tests/) are correctly formatted.

### CODE REVIEW

File: src/ethereum_test_tools/common/types.py [hunk 1]
```diff
@@ -524,7 +524,12 @@ def is_empty(self: "Account") -> bool:
             (self.nonce == 0 or self.nonce is None)
             and (self.balance == 0 or self.balance is None)
             and self.has_empty_code()
-            and (not self.storage or self.storage == {} or self.storage is None)
+            and (
+                not self.storage
+                or self.storage is None
+                or self.storage == {}
+                or not any(Storage.parse_key_value(v) != 0 for _, v in self.storage.items())
```
Comment (danceratopz) [inline src/ethereum_test_tools/common/types.py:531]: How about?
```suggestion
                or all(v in [0, None] for v in self.storage.values())
```

File: src/ethereum_test_tools/tests/test_types.py [hunk 2]
```diff
@@ -101,6 +101,57 @@ def test_storage():
...
+                nonce=0,
+                balance=0,
+                code="",
+                storage={
+                    1: 0,
+                },
+            ),
+            id="only_zero_storage_values",
+        ),
+    ],
```
Comment (danceratopz) [inline src/ethereum_test_tools/tests/test_types.py:146]:
```suggestion
    [
        pytest.param(
            Account(),
            id="no_fields",
        ),
        pytest.param(
            Account(
                nonce=0,
            ),
            id="zero_nonce",
        ),
        pytest.param(
            Account(
                nonce=None,
            ),
            id="none_nonce",
        ),
        pytest.param(
            Account(
                balance=0,
            ),
            id="zero_balance",
        ),
        pytest.param(
            Account(
                balance=None,
            ),
            id="none_balance",
        ),
        pytest.param(
            Account(
                code="",
            ),
            id="empty_code",
        ),
        pytest.param(
            Account(
                code=None,
            ),
            id="None_code",
        ),
        pytest.param(
            Account(
                storage={},
            ),
            id="empty_storage",
        ),
        pytest.param(
            Account(
                storage=None,
            ),
            id="empty_storage",
        ),
        pytest.param(
            Account(
                nonce=0,
                balance=0,
                code="",
                storage={
                    1: 0,
                },
            ),
            id="only_zero_storage_values",
        ),
        pytest.param(
            Account(
                nonce=0,
                balance=0,
                code="",
                storage={
                    1: None,
                },
            ),
            id="only_none_storage_values",
        ),
        pytest.param(
            Account(
                nonce=0,
                balance=0,
                code="",
                storage={
                    1: None,
                    2: 0,
                },
            ),
            id="zero_and_none_storage_values",
        ),
    ],
```

**Review by danceratopz — COMMENTED** [review]
Good catch! 

Just a suggestion for readability and checking `None` storage values - added a few tests for that.

There's no changelog entry. I'd suggest that if we think it's not required, we should make a comment in the pr description next to the item in the todo list. I think it's borderline in this case :slightly_smiling_face:

File: src/ethereum_test_tools/common/types.py (hunk 1 above)
Comment (marioevz) [reply src/ethereum_test_tools/common/types.py:531]: I like `any` a bit more because it aborts checking the rest of the items once any non-zero has been found, whereas `all` will always go through all the items. I'll commit the use of `.values()` though.

Comment (danceratopz) [reply src/ethereum_test_tools/common/types.py]: Good point! :+1:

**Review by danceratopz — APPROVED** [review]
Thanks for rebasing! LGTM
